Redefining ssize_t on Windows

[We define `ssize_t` when building with MSVC](https://github.com/realm/realm-core/blob/develop/src/realm/utilities.hpp#L42) but we don't check if the type hasn't already been defined first.

We should check whether `ssize_t` is already defined before defining it by using a preprocessor guard [like libuv](https://github.com/libuv/libuv/blob/v1.11.0/include/uv-win.h#L26) which (after some googling) seems to be the de-facto way to define `ssize_t` on Windows.

Right now realm-js for Node.js doesn't build in Win32 because Core and libuv try to define `ssize_t`.

The actual error is
```
3>c:\users\yavor\.node-gyp\7.10.0\include\node\uv-win.h(27): error C2371: 'ssize_t': redefinition; different basic types
3>Y:\Projects\Realm\realm-js\vendor\realm-win-ia32\include\realm/utilities.hpp(42): note: see declaration of 'ssize_t'
```

when building [`weak_realm_notifier.cpp`](https://github.com/realm/realm-object-store/blob/master/src/impl/weak_realm_notifier.cpp) which includes both Realm and libuv headers.

--- a/src/realm/utilities.hpp
+++ b/src/realm/utilities.hpp
@@ -27,20 +27,25 @@
 #include <functional>
 
 #ifdef _MSC_VER
+
 #if REALM_UWP
 struct timeval {
     long tv_sec;
     long tv_usec;
 };
 #endif
+
 #include <windows.h>
 #include <intrin.h>
+#include <BaseTsd.h>
+
+#if !defined(_SSIZE_T_) && !defined(_SSIZE_T_DEFINED)
+typedef SSIZE_T ssize_t;
+#define _SSIZE_T_
+#define _SSIZE_T_DEFINED
 #endif
 
-#if defined(_MSC_VER)
-#include <BaseTsd.h>
-typedef SSIZE_T ssize_t;
-#endif
+#endif // _MSC_VER
 
 #include <realm/util/features.h>
 #include <realm/util/assert.hpp>

